"masking temporarily disabled" >8 years ago: should we unskip or delete these tests?

https://github.com/openedx/edx-platform/blob/71e2aa7018b3c9e461080c9501394ded447c768f/xmodule/tests/test_capa_block.py#L2042-L2082

These skips were added in commit 3ef1e0f7ad0794b317b810af8f90901cba82229b.

I ran the masked tests and they failed with type errors as follows:

```python
====================================================== FAILURES ======================================================
________________________________________ ProblemBlockTest.test_rescore_unmask ________________________________________

self = <xmodule.tests.test_capa_block.ProblemBlockTest testMethod=test_rescore_unmask>

    def test_rescore_unmask(self):
        """On problem rescore, unmasked names should appear on publish."""
        block = CapaFactory.create(xml=self.common_shuffle_xml)
        get_request_dict = {CapaFactory.input_key(): 'mask_0'}
        block.submit_problem(get_request_dict)
        # On rescore, state/student_answers should use unmasked names
        with patch.object(block.runtime, 'publish') as mock_publish:
>           block.rescore_problem(only_if_higher=False)  # lint-amnesty, pylint: disable=no-member
E           AttributeError: 'ProblemBlock' object has no attribute 'rescore_problem'

xmodule/tests/test_capa_block.py:2074: AttributeError
------------------------------------------------- Captured log call --------------------------------------------------
ERROR    edx.courseware:capa_block.py:1933 Unable to gather submission metadata, it will not be included in the event.
Traceback (most recent call last):
  File "/edx/app/edxapp/edx-platform/xmodule/capa_block.py", line 1928, in get_submission_metadata_safe
    return self.get_submission_metadata(answers, correct_map)
  File "/edx/app/edxapp/edx-platform/xmodule/capa_block.py", line 1978, in get_submission_metadata
    user_visible_answer = answer_input.get_user_visible_answer(internal_answer)
  File "/edx/app/edxapp/edx-platform/xmodule/capa/inputtypes.py", line 569, in get_user_visible_answer
    return self._choices_map[internal_answer]
KeyError: 'mask_0'
_________________________________________ ProblemBlockTest.test_save_unmask __________________________________________

self = <xmodule.tests.test_capa_block.ProblemBlockTest testMethod=test_save_unmask>

    def test_save_unmask(self):
        """On problem save, unmasked data should appear on publish."""
        block = CapaFactory.create(xml=self.common_shuffle_xml)
        with patch.object(block.runtime, 'publish') as mock_publish:
            get_request_dict = {CapaFactory.input_key(): 'mask_0'}
            block.save_problem(get_request_dict)
            mock_call = mock_publish.mock_calls[0]
            event_info = mock_call[1][1]
>           assert event_info['answers'][CapaFactory.answer_key()] == 'choice_2'
E           TypeError: string indices must be integers

xmodule/tests/test_capa_block.py:2050: TypeError
_________________________________________ ProblemBlockTest.test_reset_unmask _________________________________________

self = <xmodule.tests.test_capa_block.ProblemBlockTest testMethod=test_reset_unmask>

    def test_reset_unmask(self):
        """On problem reset, unmask names should appear publish."""
        block = CapaFactory.create(xml=self.common_shuffle_xml)
        get_request_dict = {CapaFactory.input_key(): 'mask_0'}
        block.submit_problem(get_request_dict)
        # On reset, 'old_state' should use unmasked names
        with patch.object(block.runtime, 'publish') as mock_publish:
            block.reset_problem(None)
            mock_call = mock_publish.mock_calls[0]
            event_info = mock_call[1][1]
>           assert mock_call[1][0] == 'reset_problem'
E           AssertionError: assert <ProblemBlock @789D parent=None, name=None, tags=[], display_name='Blank Advanced Problem', attempts=1, attempts_befor...source_code=None, student_answers={}, submission_wait_seconds=0, use_latex_compiler=False, weight=1, xml_attributes={}> == 'reset_problem'

xmodule/tests/test_capa_block.py:2063: AssertionError
```

I think we should probably remove them as I assume the implementation might have changed for the logic that these tests are running for.

--- a/xmodule/tests/test_capa_block.py
+++ b/xmodule/tests/test_capa_block.py
@@ -2798,48 +2798,6 @@ class ProblemBlockTest(unittest.TestCase):  # lint-amnesty, pylint: disable=miss
                    ('shuffle', ['choice_3', 'choice_1', 'choice_2', 'choice_0'])
             assert event_info['success'] == 'correct'
 
-    @unittest.skip("masking temporarily disabled")
-    def test_save_unmask(self):
-        """On problem save, unmasked data should appear on publish."""
-        block = CapaFactory.create(xml=self.common_shuffle_xml)
-        with patch.object(block.runtime, 'publish') as mock_publish:
-            get_request_dict = {CapaFactory.input_key(): 'mask_0'}
-            block.save_problem(get_request_dict)
-            mock_call = mock_publish.mock_calls[0]
-            event_info = mock_call[1][1]
-            assert event_info['answers'][CapaFactory.answer_key()] == 'choice_2'
-            assert event_info['permutation'][CapaFactory.answer_key()] is not None
-
-    @unittest.skip("masking temporarily disabled")
-    def test_reset_unmask(self):
-        """On problem reset, unmask names should appear publish."""
-        block = CapaFactory.create(xml=self.common_shuffle_xml)
-        get_request_dict = {CapaFactory.input_key(): 'mask_0'}
-        block.submit_problem(get_request_dict)
-        # On reset, 'old_state' should use unmasked names
-        with patch.object(block.runtime, 'publish') as mock_publish:
-            block.reset_problem(None)
-            mock_call = mock_publish.mock_calls[0]
-            event_info = mock_call[1][1]
-            assert mock_call[1][0] == 'reset_problem'
-            assert event_info['old_state']['student_answers'][CapaFactory.answer_key()] == 'choice_2'
-            assert event_info['permutation'][CapaFactory.answer_key()] is not None
-
-    @unittest.skip("masking temporarily disabled")
-    def test_rescore_unmask(self):
-        """On problem rescore, unmasked names should appear on publish."""
-        block = CapaFactory.create(xml=self.common_shuffle_xml)
-        get_request_dict = {CapaFactory.input_key(): 'mask_0'}
-        block.submit_problem(get_request_dict)
-        # On rescore, state/student_answers should use unmasked names
-        with patch.object(block.runtime, 'publish') as mock_publish:
-            block.rescore_problem(only_if_higher=False)  # lint-amnesty, pylint: disable=no-member
-            mock_call = mock_publish.mock_calls[0]
-            event_info = mock_call[1][1]
-            assert mock_call[1][0] == 'problem_rescore'
-            assert event_info['state']['student_answers'][CapaFactory.answer_key()] == 'choice_2'
-            assert event_info['permutation'][CapaFactory.answer_key()] is not None
-
     def test_check_unmask_answerpool(self):
         """Check answer-pool question publish uses unmasked names"""
         xml = textwrap.dedent("""

